crypto/cmp/cmp_client.c

```diff
@@ -106,9 +106,12 @@ static int save_statusInfo(OSSL_CMP_CTX *ctx, OSSL_CMP_PKISI *si)
     ss = si->statusString; /* may be NULL */
     for (i = 0; i < sk_ASN1_UTF8STRING_num(ss); i++) {
         ASN1_UTF8STRING *str = sk_ASN1_UTF8STRING_value(ss, i);
+        ASN1_UTF8STRING *dup = ASN1_STRING_dup(str);
 
-        if (!sk_ASN1_UTF8STRING_push(ctx->statusString, ASN1_STRING_dup(str)))
+        if (dup == NULL || !sk_ASN1_UTF8STRING_push(ctx->statusString, dup)) {
```

Comment:
Your suggestion won't work because then we won't jump to the error path if dup is NULL.
I don't think it's worth it to split the code up to check dup first and then goto the error path, and then check sk_ASN1_UTF8STRING_push and free and goto on error.

Reply:
Yeah. There is no point in splitting the code. The *_free() calls always work with NULL arguments gracefully.

Reply:
I agree. The code is nicely succinct as it in